libs/auth-api-lib/src/lib/services/delegations.service.ts

```diff
@@ -91,78 +90,119 @@ export class DelegationsService {
           toNationalId: d.toNationalId,
           fromNationalId: d.fromNationalId,
           fromName: d.fromDisplayName,
-          isFromCompany: d.isFromCompany,
           type: DelegationType.Custom,
           provider: DelegationProvider.Custom,
         },
     )
   }
 
-  async create(delegation: DelegationDTO): Promise<Delegation | null> {
+  async create(
+    nationalId: string,
+    delegation: DelegationDTO,
+  ): Promise<Delegation | null> {
     this.logger.debug('Creating a new delegation')
     const id = uuid()
-    return await this.delegationModel.create({ id: id, ...delegation })
+    await this.delegationModel.create({
+      id: id,
+      fromNationalId: nationalId,
+      ...delegation,
+    })
+    if (delegation.scopes) {
+      this.delegationScopeService.createMany(id, delegation.scopes)
+    }
+    return this.findOne(nationalId, id)
   }
 
   async update(
+    nationalId: string,
     delegation: DelegationDTO,
     id: string,
   ): Promise<Delegation | null> {
     this.logger.debug(`Updating a delegation with id ${id}`)
-    await this.delegationModel.update({ ...delegation }, { where: { id: id } })
-    return this.findByPk(id)
+    await this.delegationModel.update(
+      { ...delegation },
+      { where: { id: id, fromNationalId: nationalId } },
+    )
+    await this.delegationScopeService.delete(id)
```

Comment:
Should only delete scopes if a new scope array is passed?

Reply:
Depends on how the client would behave. The simplest way would if those scope which are selected are sent into the request. Then we delete the existing ones and add in all those which are selected

Reply:
Right. I guess I was expecting PATCH behaviour rather than PUT behaviour. That I could pass scopes: [] to remove all scopes, and skip the scopes parameter to not modify it.

But since this is PUT, I'm happy to keep it like it is.